lighthouse-core/gather/driver.js

```diff
@@ -1217,6 +1217,41 @@ class Driver {
     return flattenedDocument.nodes ? flattenedDocument.nodes : [];
   }
 
+  /**
+   * @param {number} backendNodeId
+   * @return {Promise<string|undefined>}
+   */
+  async resolveNodeIdToObjectId(backendNodeId) {
+    try {
+      const resolveNodeResponse = await this.sendCommand('DOM.resolveNode', {backendNodeId});
```

Comment:
are there any "call `DOM.getDocument` before running this" restrictions for these two new methods?

Reply:
on the push to frontend one yes, added a note

Reply:
hmmm is this true?

a comment @Beytoven had previously written: `// DOM.getDocument is necessary for pushNodesByBackendIdsToFrontend to properly retrieve nodeIds.`

I am pretty sure we found that while it was necessary for getting valid nodeIds, it's not necessary if you're getting these objectIds

also our codebase only has 1 remaining call to getDocument and it's unrelated.

all that said, i wouldn't mind keeping a comment around to remind us of this fact if we ever try to get nodeIds again

Reply:
I can't tell if we're saying the same thing or not :)

> a comment @Beytoven had previously written: // DOM.getDocument is necessary for pushNodesByBackendIdsToFrontend to properly retrieve nodeIds.

I'm not sure where that one is, but I see @umaar 's comment in anchor elements that's probably a typo since it uses `pushNodeByPathToFrontend`, which is where I added the note in driver already.

Are you saying that `DOM.resolveNode` also requires `DOM.getDocument`? That would be surprising to me because trace-elements doesn't use it but seems to work. Happy to update the jsdoc on this function if that's wrong though.